_publications/jors/paper.tex

```diff
@@ -0,0 +1,457 @@
+outlined in this paper of using static infrastructure coupled with
+small independent 3rd party web services providing a flexible approach
+eliminating the initial prototyping and on-going maintenance required
+for new infrastructure, while allowing developers to focus on their
+unique front-end data views.
+
+This paper presents the first deployment of the PFHub framework
+including its client-side focused design, how it employs external
+services and metadata about the code base. The paper describes the
+relative ease with which other scientific groups might adapt the
+framework for their own purposes and deploy using the fully
+reproducible Nix environment~\cite{nix}.
+
+\section*{Implementation and architecture}
+
+The PFHub framework provides a template for other small scientific
+communities to host custom content and integrate data from members of
+their community. The current deployment provides a facility for
+uploading, displaying and comparing results from benchmark problems
+supporting phase-field code developers and
+practitioners~\ref{fig:pfhub_frontpage}. However, the framework and
+overall philosophy are broadly transferable to other communities with
+some custom configuration and content generation. The framework uses
+the Jekyll static website generator~\cite{jekyll} along with automated
+front-end processing to eliminate the need for a CMS~\cite{cmsfree},
+which is generally costly to maintain especially for small scientific
+communities with limited funding and staffing.  The framework relies
+on the API, WebSocket and webhook infrastructure that underpins the
+modern web and allows external services to have full-duplex
+communication between servers and browsers. In particular, PFHub
+relies on GitHub's well maintained API and webhook functionality for
+external services (such as Travis CI~\cite{travis} and
+Staticman~\cite{staticman}).
+
+The workflow for uploading benchmark results relies on third party
+tools using the following steps, illustrated in
+Figure~\ref{fig:pfhub_website}.
+\begin{enumerate}
+  \item The users are first required to upload simulation outputs to
+    an archival resource (\emph{e.g.}, Figshare\footnote{Certain
+      commercial equipment, instruments, or materials (or suppliers,
+      or software, ...) are identified in this paper to foster
+      understanding. Such identification does not imply recommendation
+      or endorsement by the National Institute of Standards and
+      Technology, nor does it imply that the materials or equipment
+      identified are necessarily the best available for the
+      purpose.\label{disclaimer}}~\cite{figshare}) configured with
+    permissive cross-origin resource sharing (CORS).
+  \item The metadata summarizing each simulation is entered into a
+    form on the website, including relevant details such as memory
+    usage, run time and links to the data archived in the first
+    step~\ref{fig:pfhub_upload}.
```

Comment:
see above comment about clarifying this is supposed to point to a figure

Reply:
done